keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/memory/InMemoryArtifactRepository.kt

```diff
@@ -38,7 +39,7 @@ class InMemoryArtifactRepository(
   private val artifacts = mutableMapOf<UUID, DeliveryArtifact>()
 
   private val approvedVersions = mutableMapOf<EnvironmentVersionsKey, MutableList<String>>()
-  private val deployedVersions = mutableMapOf<EnvironmentVersionsKey, MutableList<String>>()
+  private val deployedVersions = mutableMapOf<EnvironmentVersionsKey, MutableList<Pair<String, Instant>>>()
```

Comment:
We now track the deployment times alongside the versions so that we can fill in the `deployedAt` and `replacedAt` fields in the artifact summaries.